keyboards/dz60/keymaps/boris_burger/keymap.c

```diff
@@ -0,0 +1,87 @@
+        MO(1)  ,          KC_A   , KC_S   , KC_D   , KC_F   , KC_G   , KC_H   , KC_J   , KC_K   , KC_L   , KC_SCLN, KC_QUOT, KC_ENT ,
+        KC_LSFT,          KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   , KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_RSFT, KC_UP  , KC_INS ,
+        KC_LCTL, KC_LGUI,          KC_LALT, KC_SPC ,          KC_SPC ,          KC_SPC ,          KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
+
+    /* FN Layer
+    * ,-----------------------------------------------------------------------------------------.
+    * |  `  |  F1 |  F2 |  F3 |  F4 |  F5 |  F6 |  F7 |  F8 |  F9 | F10 | F11 | F12 |    DEL    |
+    * |-----------------------------------------------------------------------------------------+
+    * |        |RBB T|RGB M| Hue+| Hue-| Sat+| Sat-| Val+| Val-|     |     |      |     |       |
+    * |-----------------------------------------------------------------------------------------+
+    * |         | BL T| BL M| BL+ | BL- |     |     |     |     |     |     |     |             |
+    * |-----------------------------------------------------------------------------------------+
+    * |           |     |     |     |     |RESET|     |     |     |     |     |     | PgUp|     |
+    * |-----------------------------------------------------------------------------------------+
+    * |      |       |       |                                   |     |      | Home| PgDn| End |
+    * `-----------------------------------------------------------------------------------------'
+    */
+    LAYOUT_directional(
+        KC_GRV , KC_F1  , KC_F2  , KC_F3  , KC_F4  , KC_F5  , KC_F6  , KC_F7  , KC_F8  , KC_F9  , KC_F10 , KC_F11 , KC_F12 , _______, KC_DEL ,
+        _______,          RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______, _______,
+        _______,          BL_TOGG, BL_STEP, BL_INC , BL_DEC , _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, RESET  , _______, _______, _______, _______, _______, _______, KC_PGUP, _______,
+        _______, _______,          _______, _______,          _______,          _______,          _______, _______, KC_HOME, KC_PGDN, KC_END
+        ),
+};
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch(keycode) {
+        case KC_BSPC: {
+            if (record->event.pressed) {
+                const uint8_t current_mods = get_mods();
+                const uint8_t shift = current_mods & (MOD_BIT(KC_LSHIFT)|MOD_BIT(KC_RSHIFT));
+                const uint8_t ctrl = current_mods & (MOD_BIT(KC_LCTL)|MOD_BIT(KC_RCTL));
+                bspc_mods = current_mods;
+                if(shift || ctrl) {
+                    add_key(KC_DEL);
+                    add_mods(current_mods);
+                    send_keyboard_report();
+                } else {
```

Comment:
You'd want to `return false` here to denote that you've handled the keypress, and that the firmware shouldn't hit backspace.

Reply:
The `return false` is the last line of the `case KC_BSPC:`, i.e. all of the branches there actually lead to it except for the one case when Backspace was pressed without Ctrl or Shift.

Reply:
okay, just making sure.